includes/Utilities/BackgroundJobCleanup.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Utilities;
+
+use SkyVerge\WooCommerce\Facebook\Heartbeat;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class BackgroundJobCleanup
+ *
+ * Responsible for cleaning up old completed background sync jobs from SkyVerge background job system.
+ * Each job is represented by a row in wp_options table, and these can accumulate over time.
+ *
+ * Note - this is closely coupled to the SkyVerge background job system, and is essentially a patch to improve it.
+ *
+ * @see SV_WP_Background_Job_Handler
+ *
+ * @since x.x.x
+ */
+class BackgroundJobCleanup {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->add_hooks();
+	}
+
+	/**
+	 * Add hooks.
+	 */
+	public function add_hooks() {
+		// Register our cleanup routine to run on the hourly heartbeat.
+		add_action( Heartbeat::HOURLY, array( $this, 'clean_up_old_completed_options' ) );
+	}
+
+	/**
+	 * Delete old completed product sync job rows from options table.
+	 *
+	 * Logic and database query are adapted from SV_WP_Background_Job_Handler::get_jobs().
+	 *
+	 * @see SV_WP_Background_Job_Handler
+	 * @see Products\Sync\Background
+	 */
+	public function clean_up_old_completed_options() {
+		global $wpdb;
+
+		// Query notes:
+		// - Matching product sync job only (Products\Sync\Background class).
+		// - Matching "completed" status by sniffing json option value.
+		// - Order by lowest id, to delete older rows first.
+		// - Limit number of rows (periodic task will eventually remove all).
+		// Using `get_results` so we can limit number of items; `delete` doesn't allow this.
+		$deleted_rows = $wpdb->query(
+			"DELETE
+			FROM {$wpdb->options}
+			WHERE option_name LIKE 'wc_facebook_background_product_sync_job_%'
+			AND option_value LIKE '%%\"status\":\"completed\"%'
+			ORDER BY option_id ASC
+			LIMIT 10"
```

Comment:
Setting a limit of 10 for testing. I think we should bump this to delete 100 at a time before merge. Keen for feedback on this limit, and performance of the query.

Reply:
10 is OK. If you would not delete all in one go you would be slowly depleting with each heartbeat. No need to hurry, the site is not creating 10 new entries every hour.

Reply:
Any thoughts on bumping it up to 50 or 100? Or adding a filter?

I figure that for the longer-lived stores, they'll want these to be cleared up faster, and there's probably little performance difference between 10 and 100.

Reply:
This query looks relatively safe ( I have not to measure it ), bumping to 50/100 sounds OK.

Reply:
We can increase this number safely (up to 1k). I ran a few tests:

1. Generated new entries using ( ran a few times until I had ~10k records )
```php
add_action('admin_init', function() {
	global $wpdb;
	$values = '';
	$limit = 550;
	$time = time();

	for ($i = 1;$i<=$limit;$i++) {
		// Queued
		$values .= " ('wc_facebook_background_product_sync_{$time}1{$i}', '{\"requests\":{\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4209$i\":\"UPDATE\",\"p-4213$i\":\"UPDATE\"},\"id\":\"{$time}1{$i}\",\"created_at\":\"2021-05-04 17:17:05\",\"created_by\":1,\"status\":\"queued\"}', 'no'), \n";

		// Completed.
		$values .= " ('wc_facebook_background_product_sync_{$time}2{$i}', '{\"requests\":{\"p-42092$i\":\"UPDATE\",\"p-42092$i\":\"UPDATE\",\"p-420942$i\":\"UPDATE\",\"p-42092$i\":\"UPDATE\",\"p-42092$i\":\"UPDATE\",\"p-42092$i\":\"UPDATE\",\"p-42092$i\":\"UPDATE\",\"p-42132$i\":\"UPDATE\"},\"id\":\"{$time}2{$i}\",\"created_at\":\"2021-05-04 17:17:05\",\"created_by\":1,\"status\":\"completed\"}', 'no')";
		if ( $i < $limit ):$values .= ", \n";
		else:$values .= ";";endif;
	}

	$wpdb->query("
		INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`) VALUES $values");
});
```
2. Ran (**9914 total, Query took 0.0007 seconds.**)
```sql
SELECT * FROM wp_options WHERE option_name like 'wc_facebook_background%'
```
3. Deleted 10 records (**Query took 0.0699 seconds.**)
```sql
delete FROM wp_options WHERE option_name LIKE 'wc_facebook_background%' AND option_value LIKE '%\"status\":\"completed\"%' ORDER BY option_id ASC LIMIT 10

```
4. Tested with 100 (**Query took 0.0675 seconds**)
4. Increased limit to 500 (**Query took 0.0761 seconds.**)

The difference is very little, even at 500.

Reply:
Thanks for perf testing @rcstr ! I'm thinking we bump it up fairly conservatively, since most stores won't need a huge number. Stores that do have thousands of these should probably manually clean them up. Maybe we bump to 100 or 250 to keep it fairly aggressive (speed of cleanup) but still super safe.

Reply:
Now bumped to 250.